lib/pyre/tensor/repacking.h

```diff
@@ -29,20 +29,20 @@ namespace pyre::tensor {
 
     template <int N, template <typename, std::size_t> class containerT>
     struct repacking_sum<
-        pyre::grid::Canonical<N, containerT>, pyre::grid::Canonical<N, containerT>> {
-        using packing_type = pyre::grid::Canonical<N, containerT>;
+        pyre::grid::Canonical<N, int, containerT>, pyre::grid::Canonical<N, int, containerT>> {
+        using packing_type = pyre::grid::Canonical<N, int, containerT>;
     };
 
     template <int N, template <typename, std::size_t> class containerT>
     struct repacking_sum<
-        pyre::grid::Canonical<N, containerT>, pyre::grid::Symmetric<N, containerT>> {
-        using packing_type = pyre::grid::Canonical<N, containerT>;
+        pyre::grid::Canonical<N, int, containerT>, pyre::grid::Symmetric<N, containerT>> {
```

Comment:
Compact packings (namely `pyre::grid::Symmetric` and `pyre::grid::Diagonal`) should be adjusted to follow the same design as `pyre::grid::Canonical`. Shall we implement this in this PR?